File: front/src/api/ingress.ts
import { request } from '@/utils/request'
import type { ApiResponse, PageResponse } from './types'

// Ingress查询参数
export interface IngressQueryDTO {
  clusterId: string
  namespace?: string
  name?: string
  page?: number
  pageSize?: number
  keyword?: string
}

// Ingress视图对象
export interface IngressVO {
  name: string
  namespace: string
  alias?: string
  describe?: string
  ingressClassName?: string
  createTime: string
  yaml?: string
}

// Ingress API
export const ingressApi = {
  // 获取Ingress列表
  getIngressList: (params: IngressQueryDTO): Promise<ApiResponse<PageResponse<IngressVO>>> => {
    return request.get('/ingress/list', { params })
  }
}

File: internal/web/vo/ingress.go
package vo

import (
	networkingv1 "k8s.io/api/networking/v1"
	"time"
)

type IngressVO struct {
	BaseVO
	Alias            string    `json:"alias"`            // 资源别名
	Describe         string    `json:"describe"`         // 资源描述信息
	IngressClassName *string   `json:"ingressClassName"` // Ingress类名称，指定使用的Ingress控制器
	CreateTime       time.Time `json:"createTime"`       // 资源创建时间
	Yaml             string    `json:"yaml"`             // 资源的YAML配置内容
}

func Ingress2VO(ing networkingv1.Ingress) IngressVO {
	return IngressVO{
		BaseVO: BaseVO{
			Name:      ing.Name,
			Namespace: ing.Namespace,
		},
		Alias:            ing.Annotations["alias"],
		Describe:         ing.Annotations["describe"],
		IngressClassName: ing.Spec.IngressClassName,
		CreateTime:       ing.CreationTimestamp.Time,
	}
}
